apps/wallet/src/ui/app/hooks/useOwnedNFT.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    hasPublicTransfer,
+    is,
+    SuiObject,
+    getObjectOwner,
+    type SuiAddress,
+    type GetObjectDataResponse,
+} from '@mysten/sui.js';
+import { useMemo } from 'react';
+
+export function useOwnedNFT(
+    objectData: GetObjectDataResponse | null,
+    address: SuiAddress | null
+) {
+    return useMemo(() => {
+        if (
+            !objectData ||
+            !is(objectData.details, SuiObject) ||
+            !hasPublicTransfer(objectData.details)
+        )
+            return null;
+        const owner = getObjectOwner(objectData) as { AddressOwner: string };
```

Comment:
Why this type assertion? Seems like you could logically pass an NFT that isn't owned by an address and this would fail.